File: python/non-graphical/data_collector.py
#!/usr/bin/env python3
from prime_factors import prime_factors
from random_integer_generator import generator
import numpy as np
import time

def main():
	input_number = np.arange(1,1001,1)
	rng_data = []
	decomp_data = []
	for i in input_number:
		print("---------------------------")
		print(i, "inputs")
		start_time = time.clock()
		generator1 = generator(i)
		end_time = time.clock()
		print("generator finished in {} ms".format((end_time-start_time)*1000))
		rng_data.append((end_time-start_time)*1000)
		start_time = time.clock()
		prime1 = prime_factors()
		end_time = time.clock()
		print("decomposer finised in {} ms".format((end_time-start_time)*1000))
		decomp_data.append((end_time-start_time)*1000)
		print("---------------------------\n")

	with open("data/rng_data.csv", 'w') as WRITE_FILE:
		for i in input_number:
			WRITE_FILE.write(str(i) + ", " + str(rng_data[i-1]) + "\n")
	WRITE_FILE.close()

	with open("data/decomp_data.csv", 'w') as WRITE_FILE:
		for i in input_number:
			WRITE_FILE.write(str(i) + ", " + str(decomp_data[i-1]) + "\n")
	WRITE_FILE.close()

if __name__ == '__main__':
	main()

File: cpp/non-graphical/data_collector.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <fstream>
#include <random>
#include <ctime>
#include <string>
#include "prime_factors.h"
#include "random_integer_generator.h"
using namespace std;

int main(){
  int input_array[1000];
  for(int i=1; i<1001;i++){
    input_array[i-1] = i;
  }
  float rng_data[1000];
  float decomp_data[1000];

  for(int i=1; i<1001; i++){
    cout << "---------------------------" << endl;
    cout << i << " inputs " << endl;
    clock_t start_time = clock();
    random_integer_generator generator(i);
    clock_t end_time = clock();
    double time_run = double (end_time-start_time) / CLOCKS_PER_SEC * 1000;
    cout << "generator finished in " << time_run << " ms" << endl;
    rng_data[i] = time_run;

    clock_t start_time1 = clock();
    prime_factors prime_handler;
    clock_t end_time1 = clock();
    double time_run1 = double (end_time1-start_time1) / CLOCKS_PER_SEC * 1000;
    cout << "decomposer finished in " << time_run1 << " ms" << endl;
    decomp_data[i] = time_run1;
    cout << "---------------------------" << endl;

    ofstream out_file;
    out_file.open("data/rng_data.csv");
    if(!out_file.is_open()){
      cout << "file error" << endl;
      return 1;
    }
    out_file << "number of inputs, runtime of generator(in ms)\n\n";
    for(int i=0; i<1000; i++){
      out_file << i+1 << ", " << rng_data[i] << "\n";
    }
    out_file.close();
    ofstream out_file1;
    out_file1.open("data/decomp_data.csv");
    if(!out_file1.is_open()){
      cout << "file error" << endl;
      return 1;
    }
    out_file1 << "number of inputs, runtime of decomposer(in ms)\n\n";
    for(int i=0; i<1000; i++){
      out_file1 << i+1 << ", " << decomp_data[i] << "\n";
    }
    out_file1.close();
  }
  return 0;
}
